plugins/kubernetes-backend/src/service-locator/SingleTenantServiceLocator.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Entity } from '@backstage/catalog-model';
+import {
+  ClusterDetails,
+  KubernetesClustersSupplier,
+  KubernetesServiceLocator,
+  ServiceLocatorRequestContext,
+} from '../types/types';
+
+// This locator assumes that service is located on one cluster
+// Therefore it will always return specified cluster provided in backstage.io/kubernetes-cluster annotation
+// If backstage.io/kubernetes-cluster annotation not provided will always return all cluster provided
+export class SingleTenantServiceLocator implements KubernetesServiceLocator {
```

Comment:
I think `SingleTenant` is a bit of a misnomer. Shouldn't it be something like annotation filter?

Reply:
Did you meant the service locator name? To be 'AnnotationFilter'

Reply:
@mclarke47  As the docs for `multiTenant` assumes the service is located in all the provided clusters. so this method `singleTenant` - `SingleTenantServiceLocator` assume that the service is located in one cluster from the provided clusters, to identify the cluster annotation `backstage.io/kubernetes-cluster` were used. Also if you don't specify the annotation by default to go the `multiTenant` flow.

Even we can have one more tenant `hybridTenant` to accommodate both `singleTenant & multiTenant` together assumes the service is located in one or more / all.